lib/ts/ink_memory.h

```diff
@@ -89,6 +89,11 @@ void *ats_free_null(void *ptr);
 void ats_memalign_free(void *ptr);
 int ats_mallopt(int param, int value);
 
+/* use IOBufs as the underlying allocation pool */
+void *ats_malloc_pool(size_t size);
+void *ats_realloc_pool(void *ptr, size_t size);
+void ats_free_pool(void *ptr);
```

Comment:
Yeah, James is right, definitely not awesome to have cross pollination of implementations and interfaces across module boundaries :-/. Probably should go into I_IOBuffer.h ?